content/fast/reduce-network-payloads-using-text-compression/codelab-text-compression-brotli.md

```diff
@@ -0,0 +1,243 @@
+var express = require('express');
+<strong>var shrinkRay = require('shrink-ray');</strong>
+</pre>
+
+And add it as a middleware before `express.static` is mounted:
+
+<pre class="prettyprint">
+//...
+var app = express();
+
+// compress all requests
+<strong>app.use(shrinkRay());</strong>
+</pre>
+
+Now reload the app, and take a look at the bundle size in the Network panel:
+
+<img class="screenshot" src="./network-dynamic-compression-brotli.png"  alt="Bundle size with dynamic Brotli compression">
+
+You can now see `brotli` is applied from `bz` in the `Content-Encoding` header. `main.bundle.js` is reduced from **225 KB to 53.1 KB**! This is ~14% smaller compared to `gzip` (61.6 KB).
+
+<div class="aside note">
+Brotli has eleven quality levels  from “0” (no compression) to “9” (maximum compression). A quality of "1" is very fast but less effective, whereas a quality setting of "11" is very slow but provides big savings in file size. Note that unlike the standard brotli library, which defaults to quality 11, <code>shrink-ray</code> defaults to quality 4, which is generally more appropriate for dynamic content. You can adjust the parameters of the brotli algorithm by passing in an <code>options</code> parameters to <code>shrinkRay([options])</code>.
+</div>
+
+### Static compression
+
+The idea behind static compression is to have assets compressed and saved ahead of time.
+
+#### Pros:
++  Latency due to high compression levels is not a concern anymore. Nothing needs to happen on-the-fly to compress files as they can now be fetched directly.
+
+#### Cons:
++  Assets need to compressed with every build. Build times can increase significantly if high compression levels are used.
+
+#### Static compression with Node/Express and webpack
+
+Since **static compression** involves compressing files ahead of time, webpack settings can be modified to compress assets as part of the build step. The `brotli-webpack-plugin` can be used for this.
+
+Begin by adding it as a `devDependency` in `package.json`:
+
+<pre class="prettyprint">
+"devDependencies": {
+  //...
+ <strong>"brotli-webpack-plugin": "^1.1.0"</strong>
+},
+</pre>
+
+Like any other webpack plugin, import it in the configurations file, `webpack.config.js`:
+
+<pre class="prettyprint">
+const path = require("path");
+
+//...
+<strong>const BrotliPlugin = require('brotli-webpack-plugin');</strong>
+},
+</pre>
+
+And include it within the plugins array:
+
+<pre class="prettyprint">
+module.exports = {
+  // ... 
+  plugins: [
+    // ... 
+    <strong>new BrotliPlugin({
+      asset: '[file].br[query]',
+      test: /\.(js)$/
+    })</strong>
+  ]
+}
+},
+</pre>
+
+The following arguments are used in the plugin array:
++  `asset`: The target asset name. Defaults to `'[path].br[query]'`
+ +  `[file]` is replaced with the original asset file name
+ +  `[query]` is replaced with the query
++  `test`: All assets that match this RegExp (i.e. javascript assets ending in `.js`) are processed
```

Comment:
I think at the end of this block you may want to add a brief example.

"For example, a file named foo.js would be renamed to ____"